src/anomalib/metrics/per_image/pimo.py

```diff
@@ -0,0 +1,956 @@
+
+    if paths is not None:
+        _validate_is_source_images_paths(paths, expected_num_paths=anomaly_maps.shape[0])
+
+    # other validations are done in the numpy code
+
+    threshs_array, shared_fpr_array, per_image_tprs_array, _, aupimos_array, num_threshs_auc = pimo_numpy.aupimo_scores(
+        anomaly_maps_array,
+        masks_array,
+        num_threshs,
+        binclf_algorithm=binclf_algorithm,
+        fpr_bounds=fpr_bounds,
+        force=force,
+    )
+
+    # tensors are build with `torch.from_numpy` and so the returned tensors
+    # will share the same memory as the numpy arrays
+    device = anomaly_maps.device
+    # N: number of images, K: number of thresholds
+    # shape => (K,)
+    threshs = torch.from_numpy(threshs_array).to(device)
+    # shape => (K,)
+    shared_fpr = torch.from_numpy(shared_fpr_array).to(device)
+    # shape => (N, K)
+    per_image_tprs = torch.from_numpy(per_image_tprs_array).to(device)
+    # shape => (N,)
+    aupimos = torch.from_numpy(aupimos_array).to(device)
+
+    pimoresult = PIMOResult(
+        threshs=threshs,
+        shared_fpr=shared_fpr,
+        per_image_tprs=per_image_tprs,
+        paths=paths,
+    )
+    aupimoresult = AUPIMOResult.from_pimoresult(
+        pimoresult,
+        fpr_bounds=fpr_bounds,
+        # not `num_threshs`!
+        # `num_threshs` is the number of thresholds used to compute the PIMO curve
+        # this is the number of thresholds used to compute the AUPIMO integral
+        num_threshs_auc=num_threshs_auc,
+        aupimos=aupimos,
+    )
+    return pimoresult, aupimoresult
+
+
+# =========================================== TORCHMETRICS ===========================================
+
+
+class PIMO(Metric):
+    """Per-IMage Overlap (PIMO, pronounced pee-mo) curves.
+
+    This torchmetrics interface is a wrapper around the functional interface, which is a wrapper around the numpy code.
+    The tensors are converted to numpy arrays and then passed and validated in the numpy code.
+    The results are converted back to tensors and wrapped in an dataclass object.
+
+    PIMO is a curve of True Positive Rate (TPR) values on each image across multiple anomaly score thresholds.
+    The anomaly score thresholds are indexed by a (cross-image shared) value of False Positive Rate (FPR) measure on
+    the normal images.
+
+    Details: `anomalib.metrics.per_image.pimo`.
+
+    Notation:
+        N: number of images
+        H: image height
+        W: image width
+        K: number of thresholds
+
+    Attributes:
+        anomaly_maps: floating point anomaly score maps of shape (N, H, W)
+        masks: binary (bool or int) ground truth masks of shape (N, H, W)
+
+    Args:
+        num_threshs: number of thresholds to compute (K)
+        binclf_algorithm: algorithm to compute the binary classifier curve (see `binclf_curve_numpy.Algorithm`)
+
+    Returns:
+        PIMOResult: PIMO curves dataclass object. See `PIMOResult` for details.
+    """
```

Comment:
I think examples section here would be really useful. Like what do I need to run AUPIMO with some basic torch input and output